webapp/components/DataQuery/Visualizer/Table/Row/Column/ColumnHeader.js

```diff
@@ -44,7 +44,7 @@ const ColumnHeader = (props) => {
   })
 
   const nodeDefUuid = NodeDef.getUuid(nodeDef)
-  const nodeDefLabel = NodeDef.getLabel(nodeDef, lang)
+  const nodeDefLabel = itemLabelFunction(nodeDef, lang)
```

Comment:
dont we have a hook to get node def label?

Reply:
We have a selector into store/survey/state which is getNodeDefLabel

`
export const getNodeDefLabel = (nodeDef) => (state) => {
  const surveyInfo = getSurveyInfo(state)
  const langApp = I18nState.getLang(state)
  const langSurvey = Survey.getLanguage(langApp)(surveyInfo)
  return NodeDef.getLabel(nodeDef, langSurvey)
}
`

so maybe we should add a property to redux -> webapp/store/ui/ and store there the label function selected by the user and avoid to propagate through all of the components.

What do you think?

Reply:
I dont think we need to store in redux the function. we can just add a hook `webapp/store/survey/hooks/index.js` useNodeDefLabel(nodeDef)=>{...} that calls the state selector

Reply:
how do we change the function to use into the useNodeDefLabel, the local State, or the global State?

I mean now we have a switch on the UI which uses a hook which is useLabelFunctionSelector, but at the end, we need to propagate this ItemLabelFunction or store in some global Storage to share between components without the need to propagate

Reply:
I think that Ramiro talks about a place where we can store the preference about the preferred way to show the node defs, by name or by label, so Redux store looks like the perfect place for this (and maybe in the store we can "store" only the code of the selected option, not the function).

Reply:
I'm talking about the function @SteRiccio